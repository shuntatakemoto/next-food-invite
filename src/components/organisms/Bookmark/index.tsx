import React from 'react';
import { useBookmarkList } from '../../../hooks/useBookmarkList';
import FullPost from '../../molecules/FullPost';

export const Bookmark: React.FC = () => {
  const { uid, posts } = useBookmarkList();

  return (
    <div className='grid grid-cols-2 xl:grid-cols-4 text-center'>
      {posts[0]?.id && (
        <>
          {posts.map((post) => (
            <FullPost
              key={post.id}
              id={post.id}
              uid={uid}
              listId={post.listid}
              listname={post.listname}
              timestamp={post.timestamp}
              username={post.username}
              emojiname={post.emojiname}
            />
          ))}
        </>
      )}
    </div>
  );
};
