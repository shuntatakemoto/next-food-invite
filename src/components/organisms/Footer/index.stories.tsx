import { Story } from '@storybook/react';
import React from 'react';

import { Footer, FooterProps } from './index';

const FooterStory = {
  title: 'Components/Organisms/Footer',
  component: Footer,
};

export default FooterStory;

const Template: Story<FooterProps> = (args) => <Footer {...args} />;

export const LoggedIn = Template.bind({});
LoggedIn.args = { isSignedIn: true };

export const LoggedOut = Template.bind({});
LoggedOut.args = {};
