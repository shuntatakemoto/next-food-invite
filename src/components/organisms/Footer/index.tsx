import { Emoji } from 'emoji-mart';
import Link from 'next/link';
import React from 'react';

export type FooterProps = {
  isSignedIn?: boolean;
  signOut?: () => void;
};

export const Footer: React.FC<FooterProps> = ({ isSignedIn, signOut }) => {
  return (
    <div className='flex justify-evenly items-center mt-auto h-16 bg-sub-color'>
      {isSignedIn ? (
        <Link href='/' passHref>
          <p className='text-xl font-bold'>Home</p>
        </Link>
      ) : (
        <div className=''>
          <Emoji emoji='partying_face' size={32} />
        </div>
      )}

      {isSignedIn ? (
        <button onClick={signOut} className='text-xl font-bold'>
          Logout
        </button>
      ) : (
        <div className=''>
          <Emoji emoji='drooling_face' size={32} />
        </div>
      )}
    </div>
  );
};
