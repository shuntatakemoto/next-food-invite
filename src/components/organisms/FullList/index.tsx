import React from 'react';
import { Lists } from '../../../types/lists';
import { Button } from '../../atoms/Button';
import { Headline } from '../../atoms/Headline';
import FullPost from '../../molecules/FullPost';

export type FullListProps = {
  posts: Lists;
  uid: string;
  addList?: () => void;
};

const FullList: React.FC<FullListProps> = (props) => {
  return (
    <div>
      <div className='py-8'>
        <Headline headline='マイリスト' />
      </div>
      {props.posts[0]?.id ? (
        <div className='grid grid-cols-2 text-center xl:grid-cols-3'>
          {props.posts.map((post) => (
            <FullPost
              key={post.id}
              uid={props.uid}
              listId={post.id}
              listname={post.listname}
              timestamp={post.timestamp}
              username={post.username}
              emojiname={post.emojiname}
              isBookmarkPage={false}
            />
          ))}
        </div>
      ) : (
        <div className='py-8'>
          <Headline headline='まだリストがありません' size='small' />
          <div className='pt-4 text-center'>
            <Button label='リストを追加する' onClick={props.addList} size='small' />
          </div>
        </div>
      )}
    </div>
  );
};

export default FullList;
