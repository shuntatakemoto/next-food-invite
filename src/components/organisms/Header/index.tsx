import React from 'react';
// import { useSelector } from 'react-redux';
// import { selectUser } from '../../features/userSlice';
import Link from 'next/link';
import LibraryAddIcon from '@material-ui/icons/LibraryAdd';
import BookmarkIcon from '@material-ui/icons/Bookmark';
import SearchIcon from '@material-ui/icons/Search';
import { Emoji } from 'emoji-mart';

export const Header: React.FC = () => {
  const user = useSelector(selectUser);
  const createLink = `${user.uid}/create-list`;
  const searchLink = `/search-list`;
  const bookmarkLink = `${user.uid}/bookmark-list`;

  return (
    <div className='bg-sub-color flex h-16 items-center justify-evenly'>
      <Link href='/'>
        <h3 className='text-2xl font-bold'>Food Invite</h3>
      </Link>

      {user.uid ? (
        <Link href={createLink}>
          <LibraryAddIcon fontSize='large' />
        </Link>
      ) : (
        <div className=''>
          <Emoji emoji='knife_fork_plate' size={32} />
        </div>
      )}

      {user.uid ? (
        <Link href={searchLink}>
          <SearchIcon fontSize='large' />
        </Link>
      ) : (
        <div className=''>
          <Emoji emoji='male-cook' size={32} />
        </div>
      )}

      {user.uid ? (
        <Link href={bookmarkLink}>
          <BookmarkIcon fontSize='large' />
        </Link>
      ) : (
        <div className=''>
          <Emoji emoji='yum' size={32} />
        </div>
      )}
    </div>
  );
};
