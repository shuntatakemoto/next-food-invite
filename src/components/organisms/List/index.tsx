import { Emoji } from 'emoji-mart';
import React from 'react';
import Modal from 'react-modal';
import { useList } from '../../../hooks/useList';
import { Button } from '../../atoms/Button';
import { Headline } from '../../atoms/Headline';
import { Share } from '../../molecules/Share';
import { SubProfile } from '../../molecules/SubProfile';

Modal.setAppElement('#__next');

export const List: React.FC = () => {
  const {
    user,
    post,
    addLink,
    openModal,
    modalIsOpen,
    afterOpenModal,
    closeModal,
    DmLink,
    bookmark,
    deleteList,
    shareUrl,
  } = useList();

  return (
    <div className='pt-5 '>
      <SubProfile avatar={post.avatar} username={post.username} />
      <div className='mb-6'>
        <div className='text-center py-10'>
          {post.emojiname && <Emoji emoji={post.emojiname} size={64} set='twitter' />}
        </div>
        <div className='mb-8'>
          <Headline headline={post.listname} />
        </div>

        <div className='text-center '>
          <div className='mb-2'>{user.uid && <Button label='追加する' onClick={addLink} />}</div>
          <div className='mb-2'>
            {user.uid && <Button label='シェアする' onClick={openModal} />}
            <Modal
              isOpen={modalIsOpen}
              onAfterOpen={afterOpenModal}
              onRequestClose={closeModal}
              style={{
                overlay: {
                  position: 'fixed',
                  top: 0,
                  left: 0,
                  right: 0,
                  bottom: 0,
                  backgroundColor: 'rgba(255, 255, 255, 0.75)',
                },
                content: {
                  position: 'absolute',
                  top: '35%',
                  left: '40px',
                  right: '40px',
                  bottom: '40%',
                  border: '1px solid #ccc',
                  background: '#fff',
                  overflow: 'auto',
                  WebkitOverflowScrolling: 'touch',
                  borderRadius: '4px',
                  outline: 'none',
                  padding: '20px',
                },
              }}
            >
              <Share text={`${post.listname}\n`} url={shareUrl} onClick={closeModal} />
            </Modal>
          </div>
          <div className='mb-2'>
            <Button label='一緒に行きたい' onClick={DmLink} />
          </div>
          <div className='mb-2'>
            {user.uid && <Button label='ブックマークする' onClick={bookmark} />}
          </div>
          <div className=' mb-4'>
            {user.uid && (
              <Button onClick={deleteList} label='このリストを削除する' primary={true} />
            )}
          </div>
        </div>
      </div>
    </div>
  );
};
