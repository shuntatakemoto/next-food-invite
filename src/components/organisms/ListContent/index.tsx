import React, { useState, useEffect } from 'react';
import { useRouter } from 'next/dist/client/router';
import { db } from '../../../libs/firebase';
import ListContentCard from '../../molecules/ListContentCard';

export const ListContent: React.FC = () => {
  const router = useRouter();
  const { uid, listId }: any = router.query;
  const [posts, setPosts] = useState([
    {
      restaurantId: '',
      username: '',
      timestamp: null,
      imageurl: '',
      memo: '',
      url: '',
      name: '',
    },
  ]);

  useEffect(() => {
    if (uid && listId) {
      const unSub = db
        .collection(uid)
        .doc(listId)
        .collection('restaurant')
        .orderBy('timestamp', 'desc')
        .onSnapshot((snapshot: { docs: any[] }) =>
          setPosts(
            snapshot.docs.map((doc) => ({
              restaurantId: doc.id,
              username: doc.data().username,
              timestamp: doc.data().timestamp,
              imageurl: doc.data().imageurl,
              memo: doc.data().memo,
              url: doc.data().url,
              name: doc.data().name,
            })),
          ),
        );
      return () => {
        unSub();
      };
    }
  }, [uid]);

  return (
    <div className='grid grid-cols-2 xl:grid-cols-4 text-center'>
      {posts[0]?.name && (
        <>
          {posts.map((post: any) => (
            <ListContentCard
              key={post.name}
              restaurantId={post.restaurantId}
              timestamp={post.timestamp}
              username={post.username}
              imageurl={post.imageurl}
              memo={post.memo}
              url={post.url}
              name={post.name}
            />
          ))}
        </>
      )}
    </div>
  );
};
