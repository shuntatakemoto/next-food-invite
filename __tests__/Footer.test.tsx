/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';

import { Footer } from '../src/components/organisms/Footer';

const signOut = jest.fn();

test('is  not signed in', () => {
  const { asFragment } = render(<Footer />);
  expect(asFragment()).toMatchSnapshot();
});

test('is signed in', () => {
  const { asFragment } = render(<Footer isSignedIn={true} signOut={signOut} />);
  expect(asFragment()).toMatchSnapshot();
});
