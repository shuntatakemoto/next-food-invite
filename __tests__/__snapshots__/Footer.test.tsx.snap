// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is  not signed in 1`] = `
<DocumentFragment>
  <div
    class="flex justify-evenly items-center mt-auto h-16 bg-sub-color"
  >
    <div
      class=""
    >
      <span
        aria-label="ðŸ¥³, partying_face"
        class="emoji-mart-emoji"
      >
        <span
          style="width: 32px; height: 32px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 75% 14.285714285714286%;"
        />
      </span>
    </div>
    <div
      class=""
    >
      <span
        aria-label="ðŸ¤¤, drooling_face"
        class="emoji-mart-emoji"
      >
        <span
          style="width: 32px; height: 32px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 67.85714285714286% 37.5%;"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`is signed in 1`] = `
<DocumentFragment>
  <div
    class="flex justify-evenly items-center mt-auto h-16 bg-sub-color"
  >
    <p
      class="text-xl font-bold"
      href="/"
    >
      Home
    </p>
    <button
      class="text-xl font-bold"
    >
      Logout
    </button>
  </div>
</DocumentFragment>
`;
